src/env.js

```diff
@@ -46,15 +48,31 @@ export function setShimMode () {
 
 export const edge = !navigator.userAgentData && !!navigator.userAgent.match(/Edge\/\d+\.\d+/);
 
-export const baseUrl = document.baseURI;
+function getBaseURL() {
+  let baseUrl;
+
+  if (hasDocument) {
+    baseUrl = document.baseURI;
+  }
+
+  if (!baseUrl && typeof location !== 'undefined') {
+    baseUrl = location.href.split('#')[0].split('?')[0];
+    const lastSepIndex = baseUrl.lastIndexOf('/');
+    if (lastSepIndex !== -1)
+      baseUrl = baseUrl.slice(0, lastSepIndex + 1);
+  }
+
+  return baseUrl;
+}
+export const baseUrl = getBaseURL();
```

Comment:
How about something like:

```suggestion
export const baseUrl = hasDocument ? document.baseURI : `${location.protocol}//${location.host}${location.pathname.includes('/') ? location.pathname.slice(0, location.pathname.lastIndexOf('/') + 1) : location.pathname}`;
```

Since it won't work if there is no `location` anyway.